test/framework/options.py

```diff
@@ -1974,10 +1974,15 @@ def test_recursive_module_unload(self):
             self.eb_main(args, do_build=True, verbose=True)
 
             toy_module = os.path.join(self.test_installpath, 'modules', 'all', 'toy', '0.0-deps')
+
             if get_module_syntax() == 'Lua':
                 toy_module += '.lua'
+                is_loaded_regex = re.compile(r'if not \( isloaded\("gompi/2018a"\) \)', re.M)
+            else:
+                # Tcl syntax
+                is_loaded_regex = re.compile(r"if { !\[is-loaded gompi/2018a\] }", re.M)
+
             toy_module_txt = read_file(toy_module)
-            is_loaded_regex = re.compile(r"if { !\[is-loaded gompi/2018a\] }", re.M)
```

Comment:
so, with `Lua` this test wasn't actually testing? (just checking if I understand)

Reply:
Indeed. We were checking for a missing pattern using a Tcl regex pattern in a Lua module file.

That probably shows this test should be rewritten a bit, but I just wanted to fix the obvious problem when I noticed it.